subsys/llext/llext_load.c

```diff
@@ -390,14 +383,37 @@ static int llext_map_sections(struct llext_loader *ldr, struct llext *ext)
 				continue;
 			}
 
+			/*
+			 * Exported symbols region can also overlap
+			 * with rodata.
+			 */
+			if (i == LLEXT_MEM_EXPORT || j == LLEXT_MEM_EXPORT) {
+				continue;
+			}
+
+			/*
+			 * FIXME on RISC-V, GCC sometimes orders sections like this:
+			 * .rodata - .sdata - .srodata
+			 * because .sdata is writable, this causes an overlap
+			 */
+			if (IS_ENABLED(CONFIG_RISCV) && i == LLEXT_MEM_RODATA &&
+			    j == LLEXT_MEM_DATA) {
+				continue;
+			}
+
+			if (IS_ENABLED(CONFIG_RISCV) && i == LLEXT_MEM_DATA &&
+			    j == LLEXT_MEM_RODATA) {
+				continue;
+			}
```

Comment:
Not sure it's OK to ignore this overlap - depending on how RISCV manages the memory partitions, the end result might be read-only access to `.sdata`. 
Can the ordering be fixed with a linker script?

Reply:
It should be possible to re-order the sections in a linker script. 
I am not sure why this scenario only occurs on RISC-V, to be honest.

Reply:
Maybe this an idiosyncrasy of my specific compiler? I am using the pre-compiled zephyr sdk (zephyr-sdk-0.16.5-1).